apps-rendering/src/client/newsletterEmbeds.ts

```diff
@@ -0,0 +1,61 @@
+type NewsletterHeightEventType = { source: MessageEventSource; origin: string };
+type MessageType = {
+	type: string;
+	value: string;
+};
+
+const allowedOrigins = ['https://www.theguardian.com'];
+export default (): Promise<void> => {
+	const allIframes: HTMLIFrameElement[] = [].slice.call(
+		document.querySelectorAll('.email-sub__iframe'),
+	);
+
+	// Tell the iframes to resize once this script is loaded
+	// Otherwise, earlier resize events might be missed
+	// So we don't have to load this script as a priority on each load
+	allIframes.forEach((iframe) => {
+		iframe.contentWindow?.postMessage('resize', '*');
+	});
+
+	window.addEventListener('message', (event) => {
+		if (
+			!allowedOrigins.includes(
+				(event as NewsletterHeightEventType).origin,
+			)
+		)
+			return;
+
+		const iframes: HTMLIFrameElement[] = allIframes.filter((i) => {
+			try {
+				return (
+					i.contentWindow ===
+					(event as NewsletterHeightEventType).source
+				);
+			} catch (e) {
+				return false;
+			}
+		});
+		if (iframes.length !== 0) {
+			try {
+				const message: MessageType = JSON.parse(
+					event.data,
+				) as MessageType;
```

Comment:
Can we verify the structure of this data to guard against possible runtime exceptions? Perhaps `typeof message?.type === 'string'` and `typeof message?.value === 'string'`? I think that would sidestep the need to typecast?

Reply:
That's a good idea. I think `.type` should always be a string,` and `.value` could reasonably be `string` or `number`

Reply:
I found a similar implementation in https://github.com/guardian/dotcom-rendering/blob/aeb7f68d9849d3489d6cdbf78894e65c22506fcc/apps-rendering/src/client/interactives.ts, which has been useful here